restapi/src/main/java/io/stargate/web/docsapi/service/QueryExecutor.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.docsapi.service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import io.reactivex.Flowable;
+import io.reactivex.Single;
+import io.stargate.db.ImmutableParameters;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.query.builder.AbstractBound;
+import io.stargate.db.query.builder.BuiltSelect;
+import io.stargate.db.schema.Column;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/** Executes pre-built document queries, groups document rows and manages document pagination. */
+public class QueryExecutor {
+  private final Accumulator TERM = new Accumulator();
+
+  private final DataStore dataStore;
+
+  public QueryExecutor(DataStore dataStore) {
+    this.dataStore = dataStore;
+  }
+
+  public Flowable<RawDocument> queryDocs(
+      AbstractBound<?> query, int pageSize, ByteBuffer pagingState) {
+    return queryDocs(1, query, pageSize, pagingState);
+  }
+
+  public Flowable<RawDocument> queryDocs(
+      int identityDepth, AbstractBound<?> query, int pageSize, ByteBuffer pagingState) {
```

Comment:
nit: `identityDepth` is a bit confusing. This is known `keyDepth` elsewhere.

Reply:
yep, I had a few iterations on this... will clean up.

Reply:
renamed in the latest push